common/quotas/global/collection/internal/counted.go

```diff
@@ -74,21 +74,10 @@ func (c CountedLimiter) Wait(ctx context.Context) error {
 }
 
 func (c CountedLimiter) Reserve() clock.Reservation {
-	c.usage.idle.Store(0) // not idle regardless of the result
```

Comment:
tbh I think "immediately not-idle" is more technically correct, but our use shouldn't really make a difference.